syntax = "proto3";

package flwr.proto;

import "flwr/proto/fab.proto";
import "flwr/proto/run.proto";
import "flwr/proto/transport.proto";

service ClientAppIo {
    // Get Message, Context, FAB, and Run
    rpc PullClientAppInputs(
        PullClientAppInputsRequest
    ) returns (
        PullClientAppInputsResponse
    ) {}

    // Send updated Message and Context
    rpc PushClientAppOutputs(
        PushClientAppOutputsRequest
    ) returns (
        PushClientAppOutputsResponse
    ) {}	
}

message PullClientAppInputsRequest {
    sint64 token = 1;
}
message PullClientAppInputsResponse {
    Message message = 1;
    Context context = 2;
    Run run = 3;	
    // Fab fab = 4;
}
message PushClientAppOutputsRequest {
    sint64 token = 1;
    Message message = 2;
    Context context = 3;
}
message PushClientAppOutputsResponse {
    Status status = 1;
}
